components/match2/wikis/trackmania/match_summary.lua

```diff
@@ -0,0 +1,363 @@
+---
+-- @Liquipedia
+-- wiki=trackmania
+-- page=Module:MatchSummary
+--
+-- Please see https://github.com/Liquipedia/Lua-Modules to contribute
+--
+
+local Class = require('Module:Class')
+local Logic = require('Module:Logic')
+local Lua = require('Module:Lua')
+local Table = require('Module:Table')
+local VodLink = require('Module:VodLink')
+local Json = require('Module:Json')
+local Abbreviation = require('Module:Abbreviation')
+local String = require('Module:StringUtils')
+local Flags = require('Module:Flags')
+
+local DisplayHelper = Lua.import('Module:MatchGroup/Display/Helper', {requireDevIfEnabled = true})
+local MatchGroupUtil = Lua.import('Module:MatchGroup/Util', {requireDevIfEnabled = true})
+local MatchSummary = Lua.import('Module:MatchSummary/Base', {requireDevIfEnabled = true})
+local OpponentDisplay = require('Module:OpponentLibraries').OpponentDisplay
+
+local GREEN_CHECK = '[[File:GreenCheck.png|14x14px|link=]]'
+local NO_CHECK = '[[File:NoCheck.png|link=]]'
+local OVERTIME = '[[File:Cooldown_Clock.png|14x14px|link=]]'
+
+local HEADTOHEAD_PREFIX = '[[File:Match Info Stats.png|14x14px|link='
+local HEADTOHEAD_SUFFIX = '|Head to Head history]]'
+
+-- Custom Caster Class
+local Casters = Class.new(
+	function(self)
+		self.root = mw.html.create('div')
+			:addClass('brkts-popup-comment')
+			:css('white-space','normal')
+			:css('font-size','85%')
+		self.casters = {}
+	end
+)
+
+function Casters:addCaster(caster)
+	if Logic.isNotEmpty(caster) then
+		local nameDisplay = '[[' .. caster.name .. '|' .. caster.displayName .. ']]'
+		if caster.flag then
+			table.insert(self.casters, Flags.Icon(caster['flag']) .. ' ' .. nameDisplay)
+		else
+			table.insert(self.casters, nameDisplay)
+		end
+	end
+	return self
+end
+
+function Casters:create()
+	return self.root
+		:wikitext('Caster' .. (#self.casters > 1 and 's' or '') .. ': ')
+		:wikitext(table.concat(self.casters, #self.casters > 2 and ', ' or ' & '))
+end
+
+-- Custom Header Class
+local Header = Class.new(
+	function(self)
+		self.root = mw.html.create('div')
+		self.root:addClass('brkts-popup-header-dev')
```

Comment:
dev classes should not be used on live

Reply:
https://github.com/Liquipedia/Lua-Modules/pull/2312/commits/03c3c8f5f4b9abb86a942d81e26dedd3d458072a